io/zenoh-links/zenoh-link-udp/src/multicast.rs

```diff
@@ -81,14 +81,24 @@ impl LinkMulticastTrait for LinkMulticastUdp {
     }
 
     async fn write(&self, buffer: &[u8]) -> ZResult<usize> {
-        self.unicast_socket
+        match self
+            .unicast_socket
             .send_to(buffer, self.multicast_addr)
             .await
-            .map_err(|e| {
-                let e = zerror!("Write error on UDP link {}: {}", self, e);
-                tracing::trace!("{}", e);
-                e.into()
-            })
+        {
+            Ok(size) => Ok(size),
+            std::io::Result::Err(e) => {
+                if let Some(55) = e.raw_os_error() {
```

Comment:
The error code handling should be behind a platform-specific feature gate.
The code `55` means `no buffer space available` on *mac os x* (which version?) but it not guaranteed the same code is used by other platforms.